server/repository/src/db_diesel/currency.rs

```diff
@@ -0,0 +1,107 @@
+use super::{
+    currency_row::{currency, currency::dsl as currency_dsl},
+    CurrencyRow, DBType, StorageConnection,
+};
+use diesel::prelude::*;
+
+use crate::{
+    diesel_macros::{apply_equal_filter, apply_sort_no_case},
+    repository_error::RepositoryError,
+};
+
+use crate::{EqualFilter, Pagination, Sort};
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct Currency {
+    pub currency_row: CurrencyRow,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct CurrencyFilter {
+    pub id: Option<EqualFilter<String>>,
```

Comment:
Probably useful to have `is_home_currency` and `is_active` as filterable too
`is_active` doesn't appear to be used in mSupply afaics, but if we have the column, we would want to filter by that.
My suggestion:
(a) change the default to `true` and add to the filter, or
(b) remove the `is_active` column, or
(c) forget I said anything, keep the column, keep it defaulting to false and don't filter by it!